src/hera/workflows/script.py

```diff
@@ -264,6 +238,103 @@ def _build_script(self) -> _ModelScriptTemplate:
+        func_artifacts: List[Artifact],
+        output: bool,
+    ) -> Union[ModelOutputs, ModelInputs, None]:
+        if current_io is None and not func_parameters and not func_artifacts:
+            return None
+        if not func_parameters and not func_artifacts:
+            return current_io
+        if current_io is None:
+            if output:
+                current_io = ModelOutputs(
+                    parameters=[p.as_output() for p in func_parameters] or None,
+                    artifacts=[a._build_artifact() for a in func_artifacts] or None,
+                )
+            else:
+                current_io = ModelInputs(
+                    parameters=[p.as_input() for p in func_parameters] or None,
+                    artifacts=[a._build_artifact() for a in func_artifacts] or None,
+                )
+
+        already_set_params = {p.name for p in current_io.parameters or []}
+        already_set_artifacts = {a.name for a in current_io.artifacts or []}
+
+        for param in func_parameters:
+            if param.name not in already_set_params and param.name not in already_set_artifacts:
+                if current_io.parameters is None:
+                    current_io.parameters = []
+                if output:
+                    current_io.parameters.append(param.as_output())
+                else:
+                    current_io.parameters.append(param.as_input())
+
+        for artifact in func_artifacts:
+            if artifact.name not in already_set_params and artifact.name not in already_set_artifacts:
+                if current_io.artifacts is None:
+                    current_io.artifacts = []
+                current_io.artifacts.append(artifact._build_artifact())
+
+        return current_io
+
+    def _check_hera_outputs_used(self) -> bool:
+        """Check if hera outputs are used. This is needed to know if we need to create a volume."""
+        if not callable(self.source):
+            return False
+        outputs = extract_output_annotations(cast(Callable, self.source))
+        return outputs is not None
+
+    def _create_new_volume(self) -> None:
+        """Create the new volume as an EmptyDirVolume if needed."""
+        new_volume = EmptyDirVolume(name="hera__outputs_directory", mount_path=self._get_outputs_directory())
+
+        if not isinstance(self.volumes, list) and self.volumes is not None:
+            self.volumes = [self.volumes]
+        elif self.volumes is None:
+            self.volumes = []
+
+        if new_volume not in self.volumes:
+            self.volumes.append(new_volume)
+
+    def _get_outputs_directory(self) -> str:
+        """Get the outputs directory from the constructor, provide a default if not set."""
+        if isinstance(self.constructor, RunnerScriptConstructor):
+            if self.constructor.outputs_directory is not None:
+                return self.constructor.outputs_directory
+        return "/hera/outputs"
```

Comment:
This path has a leading `/` whereas in the Runner Constructor it's `outputs_directory: str = "hera/outputs"`. Could we confirm which is right? (I think with the leading?) Also, I don't think you can call this function without the constructor being a runner constructor, so you can change the `if isinstance` to `assert isinstance` to make sure linting passes, and also the outputs directory should not be able to be None, so I don't think `if self.constructor.outputs_directory is not None:` is necessary (just assert and return)